branchpro/new_posterior.py

```diff
@@ -0,0 +1,1521 @@
+    inc_key
+        label key given to the incidental data in the inc_data dataframe.
+
+    """
+    def __init__(self, inc_data, daily_serial_interval, tau, alpha, beta,
+                 time_key='Time', inc_key='Incidence Number'):
+        super(PoissonBranchProLogPosterior, self).__init__()
+
+        loglikelihood = PoissonBranchProLogLik(
+            inc_data, daily_serial_interval, tau, time_key, inc_key)
+
+        # Create a prior
+        logprior = pints.ComposedLogPrior(
+            *[pints.GammaLogPrior(alpha, beta) for _ in range(np.shape(
+                loglikelihood.cases_data)[0] - loglikelihood._tau - 1)])
+
+        self.lprior = logprior
+        self.ll = loglikelihood
+
+        # Create a posterior log-likelihood (log(likelihood * prior))
+        self._log_posterior = pints.LogPosterior(loglikelihood, logprior)
+
+    def return_loglikelihood(self, x):
+        """
+        Return the log-likelihood used for the optimisation or inference.
+
+        Parameters
+        ----------
+        x : list
+            List of free parameters used for computing the log-likelihood.
+
+        Returns
+        -------
+        float
+            Value of the log-likelihood at the given point in the free
+            parameter space.
+
+        """
+        return self.ll(x)
+
+    def return_logprior(self, x):
+        """
+        Return the log-prior used for the optimisation or inference.
+
+        Parameters
+        ----------
+        x : list
+            List of free parameters used for computing the log-prior.
+
+        Returns
+        -------
+        float
+            Value of the log-prior at the given point in the free
+            parameter space.
+
+        """
+        return self.lprior(x)
+
+    def return_logposterior(self, x):
+        """
+        Return the log-posterior used for the optimisation or inference.
+
+        Parameters
+        ----------
+        x : list
+            List of free parameters used for computing the log-posterior.
+
+        Returns
+        -------
+        float
+            Value of the log-posterior at the given point in the free
+            parameter space.
+
+        """
+        return self._log_posterior(x)
+
+    def run_inference(self, num_iter):
+        """
+        Runs the parameter inference routine for the Poisson branching process
+        model.
+
+        Parameters
+        ----------
+        num_iter : integer
+            Number of iterations the MCMC sampler algorithm is run for.
+
+        Returns
+        -------
+        numpy.array
+            3D-matrix of the proposed parameters for each iteration for
+            each of the chains of the MCMC sampler.
+
+        """
+        # Starting points using optimisation object
+        x0 = [self.lprior.mean()]*3
+        transformation = pints.RectangularBoundariesTransformation(
+            [0] * self.lprior.n_parameters(),
+            [20] * self.lprior.n_parameters()
```

Comment:
```suggestion
            [200] * self.lprior.n_parameters()
```
Rt could go above 20 for other diseases